internal/jimm/juju/model.go

```diff
@@ -728,52 +743,86 @@ func (j *JujuManager) ListModels(ctx context.Context, user *openfga.User) ([]bas
 	}
 
 	// Create map for lookup later
-	modelsMap := make(map[string]dbmodel.Model)
+	modelsMap := make(map[uint][]dbmodel.Model)
 	// Find the controllers these models reside on and remove duplicates
 	var controllers []dbmodel.Controller
 	seen := make(map[uint]bool)
 	for _, model := range models {
-		modelsMap[model.UUID.String] = model // Set map for lookup
+		models := modelsMap[model.ControllerID]
+		if models == nil {
+			models = []dbmodel.Model{}
+		}
+		models = append(models, model)
+		modelsMap[model.ControllerID] = models
+
 		if seen[model.ControllerID] {
 			continue
 		}
 		seen[model.ControllerID] = true
 		controllers = append(controllers, model.Controller)
 	}
 
-	// Call controllers for their models. We always call as admin, and we're
-	// filtering ourselves. We do this rather than send the user to be 100%
-	// certain that the models do belong to user according to OpenFGA. We could
-	// in theory rely on Juju correctly returning the models (by owner), but this
-	// is more reliable.
+	// The call to ListModels is currently broken in Juju because it always consults
+	// mongo for authorization. Because of that we have to resort to ModelInfo calls.
```

Comment:
Can we put a link to the issue we've filed against Juju? Or a link to a PR that is aiming to fix it.
I'm a bit worried that turning list models into a loop that runs ModelInfo will be prohibitively slow in a real world scenario.

Reply:
Couldn't we have this API call stick to using the `admin` user until we make the fix? That seems to be a lot easier.